src/test/java/uk/gov/companieshouse/api/testdata/service/impl/AcspProfileServiceImplTest.java

```diff
@@ -83,6 +88,140 @@ void createAcspProfileWithDefaultValues() throws DataException {
         assertEquals(0L, captured.getVersion());
     }
 
+    @Test
+    void createAcspProfileWithAmlDetails() throws DataException {
+        AcspProfileSpec spec = getAcspProfileSpec();
+
+        when(randomService.getString(8)).thenReturn("randomId");
+        AcspProfile savedProfile = getAcspProfile(spec);
+
+        when(repository.save(any(AcspProfile.class))).thenReturn(savedProfile);
+
+        AcspProfileData result = service.create(spec);
+
+        assertNotNull(result);
+        assertEquals("randomId", result.getAcspNumber());
```

Comment:
Minor but would using this value be confusing? Could we refer to `savedProfile`?
```suggestion
        assertEquals(savedProfile.getAcspNumber(), result.getAcspNumber());
```
That ensures the returned value is the one coming from db and not the necessarily the random id generated in the service (it will always match with the current code). We could use a different random value in line 135

Reply:
updated